catalog/plugin/catalog-plugin-expirationdate/src/main/java/org/codice/ddf/catalog/plugin/expiration/ExpirationDatePlugin.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    @Override
+    public DeleteRequest process(DeleteRequest deleteRequest)
+            throws PluginExecutionException, StopProcessingException {
+        return deleteRequest;
+    }
+
+    public void setOffsetFromModifedDate(int days) {
+        LOGGER.debug("Setting offset from modified date to: {}", days);
+        this.offsetFromModifiedDate = days;
+    }
+
+    public void setOverwriteIfBlank(boolean overwriteIfBlank) {
+        LOGGER.debug("Setting overwrite if blank to: {}", overwriteIfBlank);
+        this.overwriteIfBlank = overwriteIfBlank;
+    }
+
+    public void setOverwriteIfExists(boolean overwriteIfExists) {
+        LOGGER.debug("Setting overwrite if exists to: {}", overwriteIfExists);
+        this.overwriteIfExists = overwriteIfExists;
+    }
+
+    /**
+     * Updates a metacard's expiration date.
+     *
+     * @param metacard the metacard to update.
+     */
+    private void updateExpirationDate(Metacard metacard) {
+
+        Date currentExpirationDate = metacard.getExpirationDate();
+
+        // Don't overwrite expiration date if it is 'blank' and the overwriteIfBlank option is not
+        // selected.
+        if (currentExpirationDate == null && !overwriteIfBlank) {
+
+            LOGGER.debug(
+                    "The Expiration Date Pre-Ingest Plugin is not configured to overwrite 'empty' expiration dates. Not overwriting null expiration date for metacard with title [{}]. ",
+                    metacard.getTitle());
+
+            return;
+        // Don't overwrite expiration dates that are already set.
+        } else if(currentExpirationDate != null && !overwriteIfExists) {
+            LOGGER.debug(
+                    "The Expiration Date Pre-Ingest Plugin is not configured to overwrite 'existing' expiration dates. Not overwriting the existing expiration date of {} for metacard with title [{}]. ",
+                    metacard.getExpirationDate(), metacard.getTitle());
+            return;
+        }
+
+        Date newExpirationDate = calculateNewExpirationDate(metacard.getModifiedDate());
+
+        LOGGER.debug(
+                "Metacard with title [{}] has an expiration date of {}. Calculating new expiration date by adding {} day(s) to the modified date of {}. The new expiration date is {}.",
+                metacard.getTitle(), metacard.getExpirationDate(), this.offsetFromModifiedDate,
+                metacard.getModifiedDate(), newExpirationDate);
+
+        Attribute expirationDate = new AttributeImpl(Metacard.EXPIRATION, newExpirationDate);
+        metacard.setAttribute(expirationDate);
+    }
+
+    /**
+     * Calculates a new expiration date by adding offsetFromModifiedDate day(s) to the metacard's modified date.
+     *
+     * @param modifiedDate the metacard's modified date.
+     * @return the new expiration date.
+     */
+    private Date calculateNewExpirationDate(Date modifiedDate) {
+        DateTime modifiedDateTime = new DateTime(modifiedDate);
```

Comment:
Actually, unless I'm mistaken we should be using the java.time libraries over joda time.